### PR TITLE
Implementing dimenetpp

### PR DESCRIPTION
**Pull request by AnnaPicha**
Implements DimeNet++ for modelforge.

## Pull Request Summary

This PR includes:
Dimenet implementation with interaction and representation layers including
- radial bessel basis
- spherical bessel basis 

### Associated Issue(s)
 - [ ] Issue 1

## Pull Request Checklist
 - [ ] Issue(s) raised/addressed and linked
 - [ ] Includes appropriate unit test(s)
 - [ ] Appropriate docstring(s) added/updated
 - [ ] Appropriate .rst doc file(s) added/updated
 - [ ] PR is ready for review

### CODE REVIEW

File: modelforge/tests/test_dimenet.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    scripted_envelope = torch.jit.script(envelope)
+
+    # Verify that the scripted model works
+    outputs_scripted = scripted_envelope(inputs)
+    print(outputs_scripted)
+    assert torch.allclose(outputs, outputs_scripted)
+
+    # test for correct output computation
+    exponent=6 # Envelop function receives exponent = 5 but takes its increment and uses exponent = 6 -> why?
+    d_ij=0.5
```
Comment (wiederm) [inline modelforge/tests/test_dimenet.py:75]: @mwieder, please review!

File: modelforge/tests/test_dimenet.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    u_05 = 1 - (exponent+1)*(exponent+2)/2*d_ij**exponent + exponent*(exponent+2)*d_ij**(exponent+1)- exponent*(exponent+1)/2*d_ij**(exponent+2)
+    u_05/=d_ij # this should not be done, but the Envelope function does this
+    u_05 = torch.tensor([u_05], dtype=torch.float32)
```
Comment (wiederm) [inline modelforge/tests/test_dimenet.py:78]: @AnnaPicha, can we double check if this appears also in the paper

Comment (AnnaPicha) [reply modelforge/tests/test_dimenet.py:93]: @wiederm I do not really understand why this is done. I cannot find a corresponding formula in the paper. The envelope function does not seem to need normalization, it rather seems to expect normalized distances ((u(0))=1 and u(1)=0). But if, why is it normalized with d_ij? If distances get normalized, shouldn't they be normalized with the cutoff c?

File: modelforge/potential/dimenet.py [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+        self.frequencies = nn.Parameter(frequencies)  # Trainable parameter
+
+    def forward(self, d_ij: torch.Tensor) -> torch.Tensor:
+        # d_ij: Pairwise distances between atoms. Shape: (nr_pairs, 1)
+
+        # Scale distances
+        d_scaled = d_ij * self.inv_cutoff  # Shape: (nr_pairs, 1)
+
+        # Apply envelope
+        d_cutoff = self.envelope(d_scaled)  # Shape: (nr_pairs, 1)
```
Comment (AnnaPicha) [inline modelforge/potential/dimenet.py:183]: @wiederm This matches what we saw (and was commented) in the implementation of the envelope function, distances are normalized with the cutoff distance. Please check my commentary in the envelope function.

File: modelforge/potential/dimenet.py [hunk 4]
```diff
@@ -0,0 +1,359 @@
...
+        self.register_buffer(
+            "inv_cutoff", torch.tensor(1.0 / radial_cutoff, dtype=torch.float32)
+        )
+        self.envelope = Envelope(envelope_exponent)
+
+        # Initialize frequencies at canonical positions
+        frequencies = torch.pi * torch.arange(
+            1, number_of_radial_bessel_functions + 1, dtype=torch.float32
+        )
+        self.frequencies = nn.Parameter(frequencies)  # Trainable parameter
```
Comment (AnnaPicha) [inline modelforge/potential/dimenet.py:174]: @wiederm what exactly should be trainable here? number of radial bessel functions? this should be pi * n.

File: modelforge/potential/dimenet.py [hunk 5]
```diff
@@ -0,0 +1,359 @@
...
+        # d_ij: Pairwise distances between atoms. Shape: (nr_pairs, 1)
+
+        # Scale distances
+        d_scaled = d_ij * self.inv_cutoff  # Shape: (nr_pairs, 1)
+
+        # Apply envelope
+        d_cutoff = self.envelope(d_scaled)  # Shape: (nr_pairs, 1)
+
+        # Compute Bessel basis
+        basis = d_cutoff * torch.sin(
```
Comment (AnnaPicha) [inline modelforge/potential/dimenet.py:186]: @wiederm I am not sure where the factor sqrt(2/c) went missing. Apart from this and when or why normalization of the distances is done, this should be fine.

File: modelforge/tests/test_dimenet.py (hunk 1 above)
Comment (wiederm) [reply modelforge/tests/test_dimenet.py]: we have resolved this! the exponent is now not incremented but is used as provided

File: modelforge/potential/dimenet.py (hunk 5 above)
Comment (wiederm) [reply modelforge/potential/dimenet.py:187]: added! thanks for noticing!
